pysteps/blending/steps.py

```diff
@@ -809,23 +825,80 @@ def forecast(
             if measure_time:
                 starttime = time.time()
 
+            if precip_models_cascade is not None:
+                decomp_precip_models = list(precip_models_cascade[:, t])
+            else:
+                if precip_models.shape[0] == 1:
+                    decomp_precip_models = [
+                        decompositor(
+                            precip_models[0, t, :, :],
+                            bp_filter=bp_filter,
+                            fft_method=fft,
+                            output_domain=domain,
+                            normalize=True,
+                            compute_stats=True,
+                            compact_output=True,
+                        )
+                    ]
```

Comment:
Have you timed how long this takes approx?

Reply:
Yes, this gives different results on different machines, but locally I remember it took 10 seconds for 144 timesteps, for one model member. I added the multithreading option below to allow multiple members to be processed in parallel, so that you should be able to do this in 10 seconds regardless of your model size. As I said, operationally at KNMI these 10 seconds seem to hold true